Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -418,7 +418,8 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
                               didPassVisualInstructionPoint: firstInstruction,
                               routeProgress: navigationService.routeProgress)
         }
-        navigationMapView?.simulatesLocation = (navigationService.locationManager is SimulatedLocationManager) || (navigationService.locationManager is ReplayLocationManager)
+
+        navigationMapView?.simulatesLocation = navigationService.locationManager.simulatesLocation || (navigationService.simulationMode != .never)
```

Comment:
@ShanMa1991 pointed out that `.onPoorGPS` would require this setting to change dynamically. Does NavigationViewController get notified somehow when simulation mode kicks in due to poor GPS?

Reply:
Yes, I'll add the `navigationViewController.navigationService(_ :willBeginSimulating: becauseOf:)` and  `navigationViewController.navigationService(_ :willEndSimulating: becauseOf:)` to notify the `navigationMapView.simulatesLocation`